### PR TITLE
#28 support automatic renewal of selfsigned certs

### LINKED ISSUE #28: Automatic renewal of selfsigned certificates on FQDN change
Whenever the FQDN is changed, selfsigned certificates should be automatically regenerated for the new FQDN.

### PR DESCRIPTION
**Pull request by jelemux**
Resolves #28

### CODE REVIEW

File: controllers/ssl/api.go [hunk 1]
```diff
@@ -13,20 +12,9 @@ const endpointPostGenerateSSL = "/api/v1/ssl"
...
-	// private key as string.
-	GenerateSelfSignedCert(fqdn string, domain string, certExpireDays int, country string,
-		province string, locality string, altDNSNames []string) (string, string, error)
-}
-
-type cesSSLWriter interface {
-	// WriteCertificate writes the type, cert and key to the global config
-	WriteCertificate(certType string, cert string, key string) error
+type selfSignedCertificateCreator interface {
+	CreateAndSafeCertificate(certExpireDays int, country string,
```
Comment (jelemux) [inline controllers/ssl/api.go:16]:
```suggestion
	// CreateAndSafeCertificate generates and writes the type, cert and key to the global config.
	CreateAndSafeCertificate(certExpireDays int, country string,
```

File: controllers/ssl/creator.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+// NewCreator generates and writes selfsigned certificates to the ces registry.
+func NewCreator(globalConfig globalConfig) *creator {
+	return &creator{
+		globalConfig: globalConfig,
+		sslGenerator: ssl.NewSSLGenerator(),
+		sslWriter:    NewSSLWriter(globalConfig),
+	}
+}
+
+func (c *creator) CreateAndSafeCertificate(certExpireDays int, country string,
```
Comment (jelemux) [inline controllers/ssl/creator.go:40]:
```suggestion
// CreateAndSafeCertificate generates and writes the type, cert and key to the global config.
func (c *creator) CreateAndSafeCertificate(certExpireDays int, country string,
```

**Review by jelemux — COMMENTED** [review]
Some doc comments would be nice.

**Review by kahoona77 — APPROVED** [review]
lgtm

**Pull request merged**